### DISCUSSION TITLE
Sometimes connection to Radiostation failed

### DISCUSSION THREAD

**Discussion by Scynd** [schreibfaul1/ESP32-audioI2S#151] [Q&A]
Hi schreibfaul!
Sometimes the when the ESP32 trys to connect to a radio station, the connection fails. Is there a way to detect this for an automatic reconnection? I'm not as good as coding as i want to, so i don't understand everything in your code...

Cheers
Scynd

**Comment by schreibfaul1** [comment]
Hello,
if a connection to a radio station has been successfully established, connecttohost() returns true. The query could look like this:
```
if (!audio.connecttohost(... url ...)) {
    log_i ("connection failed!'); // do something else
}
```
Or sometime later:
```
if(!audio.isRunning()){
   // do something else
}
```

**Reply by Scynd** [reply]
Hi schreibfaul!
it works like a charm :)
Than you!
